### PR TITLE
Add support for custom webhook headers

### PR DESCRIPTION
**Pull request by PatrickRice-KSC**
With 17.1, GitLab now supports the ability to set custom headers on webhooks. 

See the API documentation here: 

Project:
Add/Edit: https://docs.gitlab.com/ee/api/projects.html#edit-project-hook
Set Header: https://docs.gitlab.com/ee/api/projects.html#set-a-custom-header
Delete Header: https://docs.gitlab.com/ee/api/projects.html#delete-a-custom-header

Group:
Add/Edit: https://docs.gitlab.com/ee/api/groups.html#add-group-hook
Set Header: https://docs.gitlab.com/ee/api/groups.html#set-a-custom-header
Delete Header: https://docs.gitlab.com/ee/api/groups.html#delete-a-custom-header

Note that the `value` of the header will never come back from the API, you can only ever _set_ it (with these APIs). 

Note: Ignore the branch name please; I was intending to submit Group hooks as a separate PR, but just ended up adding it here so they could share the underlying structs in a couple cases :)

### CODE REVIEW

**Comment by PatrickRice-KSC** [comment]
@svanharmelen - This PR should be ready for review when you have a moment!

File: group_hooks.go [hunk 1]
```diff
@@ -26,29 +26,30 @@ import (
...
+	DeploymentEvents          bool               `json:"deployment_events"`
+	ReleasesEvents            bool               `json:"releases_events"`
+	SubGroupEvents            bool               `json:"subgroup_events"`
+	MemberEvents              bool               `json:"member_events"`
+	EnableSSLVerification     bool               `json:"enable_ssl_verification"`
+	AlertStatus               string             `json:"alert_status"`
+	CreatedAt                 *time.Time         `json:"created_at"`
+	CustomWebhookTemplate     string             `json:"custom_webhook_template"`
+	ResourceAccessTokenEvents bool               `json:"resource_access_token_events"`
+	CustomHeaders             []HookCustomHeader `url:"custom_headers,omitempty" json:"custom_headers,omitempty"`
```
Comment (svanharmelen) [inline group_hooks.go:52]:
```suggestion
	CustomHeaders             []*HookCustomHeader `url:"custom_headers,omitempty" json:"custom_headers,omitempty"`
```

File: group_hooks.go [hunk 2]
```diff
@@ -108,26 +109,27 @@ func (s *GroupsService) GetGroupHook(pid interface{}, hook int, options ...Reque
...
+	WikiPageEvents            *bool               `url:"wiki_page_events,omitempty"  json:"wiki_page_events,omitempty"`
+	DeploymentEvents          *bool               `url:"deployment_events,omitempty" json:"deployment_events,omitempty"`
+	ReleasesEvents            *bool               `url:"releases_events,omitempty" json:"releases_events,omitempty"`
+	SubGroupEvents            *bool               `url:"subgroup_events,omitempty" json:"subgroup_events,omitempty"`
+	MemberEvents              *bool               `url:"member_events,omitempty" json:"member_events,omitempty"`
+	EnableSSLVerification     *bool               `url:"enable_ssl_verification,omitempty"  json:"enable_ssl_verification,omitempty"`
+	Token                     *string             `url:"token,omitempty" json:"token,omitempty"`
+	ResourceAccessTokenEvents *bool               `url:"resource_access_token_events,omitempty" json:"resource_access_token_events,omitempty"`
+	CustomWebhookTemplate     *string             `url:"custom_webhook_template,omitempty" json:"custom_webhook_template,omitempty"`
+	CustomHeaders             []*HookCustomHeader `url:"custom_headers,omitempty" json:"custom_headers,omitempty"`
```
Comment (svanharmelen) [inline group_hooks.go:132]:
```suggestion
	CustomHeaders             *[]*HookCustomHeader `url:"custom_headers,omitempty" json:"custom_headers,omitempty"`
```

File: group_hooks.go [hunk 3]
```diff
@@ -159,26 +161,27 @@ func (s *GroupsService) AddGroupHook(gid interface{}, opt *AddGroupHookOptions,
...
+	WikiPageEvents            *bool               `url:"wiki_page_events,omitempty" json:"wiki_page_events,omitempty"`
+	DeploymentEvents          *bool               `url:"deployment_events,omitempty" json:"deployment_events,omitempty"`
+	ReleasesEvents            *bool               `url:"releases_events,omitempty" json:"releases_events,omitempty"`
+	SubGroupEvents            *bool               `url:"subgroup_events,omitempty" json:"subgroup_events,omitempty"`
+	MemberEvents              *bool               `url:"member_events,omitempty" json:"member_events,omitempty"`
+	EnableSSLVerification     *bool               `url:"enable_ssl_verification,omitempty" json:"enable_ssl_verification,omitempty"`
+	Token                     *string             `url:"token,omitempty" json:"token,omitempty"`
+	ResourceAccessTokenEvents *bool               `url:"resource_access_token_events,omitempty" json:"resource_access_token_events,omitempty"`
+	CustomWebhookTemplate     *string             `url:"custom_webhook_template,omitempty" json:"custom_webhook_template,omitempty"`
+	CustomHeaders             []*HookCustomHeader `url:"custom_headers,omitempty" json:"custom_headers,omitempty"`
```
Comment (svanharmelen) [inline group_hooks.go:184]:
```suggestion
	CustomHeaders             *[]*HookCustomHeader `url:"custom_headers,omitempty" json:"custom_headers,omitempty"`
```

File: projects.go [hunk 4]
```diff
@@ -1197,31 +1197,41 @@ type ProjectMember struct {
...
+	JobEvents                 bool               `json:"job_events"`
+	PipelineEvents            bool               `json:"pipeline_events"`
+	WikiPageEvents            bool               `json:"wiki_page_events"`
+	DeploymentEvents          bool               `json:"deployment_events"`
+	ReleasesEvents            bool               `json:"releases_events"`
+	EnableSSLVerification     bool               `json:"enable_ssl_verification"`
+	CreatedAt                 *time.Time         `json:"created_at"`
+	ResourceAccessTokenEvents bool               `json:"resource_access_token_events"`
+	CustomWebhookTemplate     string             `json:"custom_webhook_template"`
+	CustomHeaders             []HookCustomHeader `json:"custom_headers"`
```
Comment (svanharmelen) [inline projects.go:1234]:
```suggestion
	CustomHeaders             []*HookCustomHeader `json:"custom_headers"`
```

File: projects.go [hunk 5]
```diff
@@ -1284,24 +1294,25 @@ func (s *ProjectsService) GetProjectHook(pid interface{}, hook int, options ...R
...
+	PushEvents                *bool               `url:"push_events,omitempty" json:"push_events,omitempty"`
+	PushEventsBranchFilter    *string             `url:"push_events_branch_filter,omitempty" json:"push_events_branch_filter,omitempty"`
+	ReleasesEvents            *bool               `url:"releases_events,omitempty" json:"releases_events,omitempty"`
+	TagPushEvents             *bool               `url:"tag_push_events,omitempty" json:"tag_push_events,omitempty"`
+	Token                     *string             `url:"token,omitempty" json:"token,omitempty"`
+	URL                       *string             `url:"url,omitempty" json:"url,omitempty"`
+	WikiPageEvents            *bool               `url:"wiki_page_events,omitempty" json:"wiki_page_events,omitempty"`
+	ResourceAccessTokenEvents *bool               `url:"resource_access_token_events,omitempty" json:"resource_access_token_events,omitempty"`
+	CustomWebhookTemplate     *string             `url:"custom_webhook_template,omitempty" json:"custom_webhook_template,omitempty"`
+	CustomHeaders             []*HookCustomHeader `url:"custom_headers,omitempty" json:"custom_headers,omitempty"`
```
Comment (svanharmelen) [inline projects.go:1315]:
```suggestion
	CustomHeaders             *[]*HookCustomHeader `url:"custom_headers,omitempty" json:"custom_headers,omitempty"`
```

File: projects.go [hunk 6]
```diff
@@ -1334,24 +1345,25 @@ func (s *ProjectsService) AddProjectHook(pid interface{}, opt *AddProjectHookOpt
...
```
Comment (svanharmelen) [inline projects.go:1366]:
```suggestion
	CustomHeaders             *[]*HookCustomHeader `url:"custom_headers,omitempty" json:"custom_headers,omitempty"`
```

**Review by svanharmelen — CHANGES_REQUESTED** [review]
I acknowledge this feels weird `*[]*HookCustomHeader`, but it allows you to make a difference between an empty slice and no value set at all which in some API is needed. So for consistence sake I want to use this pattern in all option structs.

**Comment by PatrickRice-KSC** [comment]
@svanharmelen  - No problem at all, I completely understand! All changes are made 😄

**Review by svanharmelen — APPROVED** [review]
Thanks @PatrickRice-KSC 👍🏻

**Pull request merged**